app/uk/gov/hmrc/economiccrimelevyregistration/models/audit/RegistrationNotLiableAuditEvent.scala

```diff
@@ -17,9 +17,17 @@
 package uk.gov.hmrc.economiccrimelevyregistration.models.audit
 
 import play.api.libs.json.{JsValue, Json, OFormat}
-import uk.gov.hmrc.economiccrimelevyregistration.models.Registration
 
-case class RegistrationNotLiableAuditEvent(registrationData: Registration) extends AuditEvent {
+sealed trait NotLiableReason
+
+object NotLiableReason {
+  case object DidNotCarryOutAmlRegulatedActivity extends NotLiableReason
+  case object SupervisedByGamblingCommission extends NotLiableReason
+  case object SupervisedByFinancialConductAuthority extends NotLiableReason
+  case object RevenueDoesNotMeetThreshold extends NotLiableReason
+}
+
+case class RegistrationNotLiableAuditEvent(internalId: String, notLiableReason: String) extends AuditEvent {
```

Comment:
Is there any reason why notLiableReason is of type String and not of type NotLiableReason (and let the writes take care of converting to a string rather than the caller performing toString on the reason)? The reason (pardon the pun!) I'm asking is it would be possible currently to construct a RegistrationNotLiableAuditEvent with a reason that is not one of the constrained reasons.

Reply:
It's a good question and one I pondered with myself, my thinking was that we never 'read' audit events so therefore we're only interested in 'writing' them, I could define just a writes for the type which would address your point, which I think i'll do for consistency else where.

I also want to add the other data where the revenue doesn't meet threshold (e.g. ap length, revenue etc.)

leave it with me

Reply:
@TonyCOgilvie I've pushed a commit which addresses your comment (hopefully!)